---
title: "Phylogenetic analysis and comparison to models"
author: "Erick Matsen and Trevor Bedford"
date: "April 24, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R setup

Install [apTreeshape](https://cran.r-project.org/web/packages/apTreeshape/index.html).
```{r eval=FALSE}
install.packages("apTreeshape")
```

## Doing sequence alignment

First we need to download the raw sequence data.
If you haven't gotten a zip file for the whole repository, visit the [data page for this exercise](https://github.com/blab/mitii/tree/master/phylogenetics/data) and download the data sets.
If you are having trouble with that, simply click [here](https://raw.githubusercontent.com/blab/mitii/master/phylogenetics/data/HIV1_REF_2010_gag_DNA.fasta) and save that page to a file.

Next, visit <http://wasabiapp.org>.
Click `Create account` to make a 1-day account (you don't need to insert your email address.)
Click on `Data -> Import` to upload your data.
To align, click on `Tools -> PRANK aligner`.

Once you have your aligned sequences, you can export them using `Data -> Export`.
*Important:* export your sequences in `phylip` format.


## Building a tree

We will do tree building using <http://www.atgc-montpellier.fr/phyml/>, the ATGC bioinformatics platform.
Upload your data, enter your email address, and hit `Execute`.

Once that completes, hit `Click to visualize your infered tree`.
You can interactively play with how the tree is displayed, then download your tree in [Newick format](https://en.wikipedia.org/wiki/Newick_format).


## Playing with your tree in R

```{r, message=FALSE}
library(apTreeshape)
```

Load the tree into R with the `read.tree` function of [ape](https://cran.r-project.org/web/packages/ape/):
```{r}
t = read.tree('processed/measles.nwk')
```

We can see that the tree is unrooted, which is the norm for doing phylogenetic analysis with reversible models.

The next step is to root the tree at a specific leaf:
```{r}
r = root(t, "AF410972|Montreal/CAN|1979", resolve.root = TRUE)
plot(r, show.tip.label=FALSE)
```

Now we can calculate the Colless tree shape statistic for our inferred tree:

```{r}
true_colless = colless(as.treeshape(r))
true_colless
```

Now we can compare it with 1000 simulations of a neutral process:

```{r}
n_sims = 1000
sim_colless = sapply(rtreeshape(n_sims, length(r$tip.label), model="yule"), colless)
hist(sim_colless)
```

Does the observed imbalance appear unusual?
For a more quantitative comparison:

```{r}
sum(true_colless > sim_colless) / n_sims
```


## Exercise: repeat tree shape analysis with flu tree

1. Load flu tree (`processed/flu.nwk`) into R
2. Pick an appropriate sequence with which to root the tree
3. Make a tree shape object and calculate the Colless statistic
4. How does it compare to the simulated trees?
