<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="author" content="Erick Matsen">
  <title>Introduction to phylogenetics</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../reveal/css/reveal.min.css">
  <link rel="stylesheet" href="../reveal/css/theme/trvrb.css" id="theme">
  <link rel="stylesheet" href="../reveal/css/font-awesome/css/font-awesome.min.css">

  <!-- For syntax highlighting -->
  <link rel="stylesheet" href="../reveal/lib/css/zenburn.css">

  <!--[if lt IE 9]>
  <script src="reveal.js/lib/js/html5shiv.js"></script>
  <![endif]-->
</head>
<body>
  <div class="reveal">
    <div class="slides">

<section>
  <h1>Introduction to phylogenetics</h1>
  <h3>Erick Matsen</h3>
  <h4>(With a great exercise from Trevor)</h4>
</section>

<section id="what-is-phylogenetics" class="slide level1">
<h1>What is phylogenetics?</h1>
</section>
<section id="section" class="slide level1">
<h1></h1>
<p><img width="970" src="figures/phylo/what_is_phylo_nofonts.svg"></p>
</section>
<section id="section-1" class="slide level1">
<h1></h1>
<p><img width="970" src="figures/phylo/what_is_phylo_taxa_nofonts.svg"></p>
</section>
<section id="section-2" class="slide level1">
<h1></h1>
<p><img width="970" src="figures/phylo/what_is_phylo_unrooted_nofonts.svg"></p>
</section>
<section id="data-sequence-alignment" class="slide level1">
<h3>“data”: sequence alignment</h3>
<p><img src="figures/phylo/bigProtAln.png" class="stretch"></p>
</section>
<section id="sequence-alignment-is-hard" class="slide level1">
<h3>Sequence alignment is hard!</h3>
<p><img src="figures/phylo/alignment-and-phylogeny-protpal-truth-only.png" height="620"></p>
</section>
<section id="sometimes-its-more-or-less-impossible" class="slide level1">
<h3>Sometimes it’s more or less impossible</h3>
<p><img src="figures/phylo/alignment-and-phylogeny-protpal-hard-case.png" height="620"></p>
</section>
<section id="phylogenetic-approach-is-best" class="slide level1">
<h3>Phylogenetic approach is best</h3>
<img width="970" src="figures/phylo/alignment-and-phylogeny-protpal-tractable-case.png">
<div class="citation">
<a href="http://dx.doi.org/10.1371/journal.pone.0034572">Westesson et al. 2012</a>
</div>
</section>
<section id="muscle-vs-prank" class="slide level1">
<h3>Muscle vs PRANK</h3>
<img width="970" src="figures/phylo/alignment-and-phylogeny-prank.png">
<div class="citation">
<p><a href="http://dx.doi.org/10.1126/science.1158395">Löytynoja and Goldman 2008</a></p>
</div>
</section>
<section id="bali-phy" class="slide level1">
<h3><a href="http://www.bali-phy.org/">BAli-Phy</a></h3>
<p>Co-estimate tree &amp; alignment using Bayesian methods <img width="970" src="figures/phylo/bali-phy.png"></p>
</section>
<section id="multiple-sequence-alignment-summary" class="slide level1">
<h3>Multiple sequence alignment: summary</h3>
<ul>
<li>Use PRANK, perhaps through <a href="http://wasabiapp.org/">Wasabi</a></li>
<li>If you need a faster algorithm and are happy with a less good result, try <a href="http://drive5.com/muscle/">Muscle</a></li>
<li>If you really care about alignments (and trees) use BAli-Phy.</li>
</ul>
</section>
<section id="types-of-phylogenetic-inference-methods" class="slide level1">
<h3>Types of phylogenetic inference methods</h3>
<ul>
<li>Distance-based</li>
<li>Parsimony</li>
<li>Likelihood-based
<ul>
<li>Maximum likelihood</li>
<li>Bayesian</li>
</ul></li>
</ul>
</section>
<section id="distance-based-phylogenetics" class="slide level1">
<h3>Distance-based phylogenetics</h3>
<p><img width="970" src="figures/phylo/what_is_phylo_dist_nofonts.svg"></p>
</section>
<section id="parsimony-phylogenetics" class="slide level1">
<h3>Parsimony phylogenetics</h3>
<p><img src="figures/phylo/tree_parsimony_nofonts.svg" height="620"></p>
</section>


				<section>
					<h3>
					<a href="https://en.wikipedia.org/wiki/Maximum_parsimony_(phylogenetics)">Parsimony</a> is
					based on <a href="https://en.wikipedia.org/wiki/Occam's_razor">Occam's razor</a></h3>
					<p>
					Among competing
					hypotheses that predict equally well, the one with the fewest assumptions should be selected.</p>

                    <br>

                    (The next few slides are from Trevor.)
				</section>

				<section>
					<h3>Parsimony suggests this topology requires 3 mutations at minimum</h3>
					<img class="stretch" src="figures/parsimony_4.png">
				</section>

				<section>
					<h3>Parsimony suggests both topologies equally tenable</h3>
					<img class="stretch" src="figures/parsimony_5.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more parsimonious?</i></h3>
					<img class="stretch" src="figures/parsimony_exercise_1.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more parsimonious?</i></h3>
					<img class="stretch" src="figures/parsimony_exercise_2.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more parsimonious?</i></h3>
					<img class="stretch" src="figures/parsimony_exercise_3.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more parsimonious?</i></h3>
					<img class="stretch" src="figures/parsimony_exercise_4.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more parsimonious?</i></h3>
					<img class="stretch" src="figures/parsimony_exercise_5.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more parsimonious?</i></h3>
					<img class="stretch" src="figures/parsimony_exercise_6.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more parsimonious?</i></h3>
					<img class="stretch" src="figures/parsimony_exercise_7.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more parsimonious?</i></h3>
					<img class="stretch" src="figures/parsimony_exercise_8.png">
				</section>


<section id="likelihood-setup" class="slide level1">
<h3>Likelihood setup</h3>
<ul>
<li>Come up with a <em>statistical model</em> of experiment</li>
<li>Parametrize that model</li>
<li>Evaluate likelihood under various parameter values</li>
</ul>
</section>
<section id="example-picking-peaches" class="slide level1">
<h3>Example: picking peaches</h3>
<p>Say that, after harvesting 20 peaches, we have 6 ripe ones.</p>
<p>Model using the <a href="http://en.wikipedia.org/wiki/Binomial_distribution">binomial distribution</a>. Say <span class="math inline">\(p\)</span> is the probability of getting a ripe peach, and each draw is independent.</p>
<p> </p>
<p>The <em>likelihood</em> of getting the observed result is <span class="math display">\[
{ {20} \choose 6} \, p^6 \, (1-p)^{20-6}.
\]</span> Recall: <span class="math inline">\({ {20} \choose 6}\)</span> is the number of ways of choosing 6 items out of 20.</p>
</section>
<section id="peach-picking-likelihood-surface" class="slide level1">
<h3>Peach picking likelihood surface</h3>
<p><img src="figures/like_intro/ripe_likelihood_nofonts.svg" height="450"></p>
<p>The <em>maximum likelihood estimate</em> of the parameter(s) of interest is the parameter value(s) that maximize the likelihood.</p>
</section>
<section id="likelihood-recap" class="slide level1">
<h3>Likelihood recap</h3>
<ul>
<li>Maximum likelihood is a way of inferring unknown parameters</li>
<li>To apply likelihood, we need a <em>model</em> of the system under investigation</li>
<li>In general, the “likelihood” is the likelihood of generating the data under the given parameters, written <span class="math inline">\(P(D | \theta),\)</span> where <span class="math inline">\(D\)</span> is the data and <span class="math inline">\(\theta\)</span> are the parameters.</li>
</ul>
</section>
<section id="setup-for-likelihood-based-phylogenetics" class="slide level1">
<h3>Setup for likelihood based phylogenetics</h3>
<p>The phylogenetic likelihood of a tree is the likelihood of generating the observed data given that tree (under the sequence evolution model).</p>
<p><img width="970" src="figures/phylo/what_is_phylo_ML.scour_nofonts.svg"></p>
<p>Note that the UW’s own <a href="http://evolution.genetics.washington.edu/phylip/felsenstein.html">Joe Felsenstein</a> was the <a href="http://www.springerlink.com/content/g2202t346n826461/">first</a> to formalize this and develop efficient algorithms.</p>
</section>
<section id="sequence-evolution-models-tell-us-the-probability-of-seeing-a-certain-mutation-in-some-period-of-evolutionary-time" class="slide level1">
<h3><a href="http://en.wikipedia.org/wiki/Substitution_model">Sequence evolution models</a> tell us the probability of seeing a certain mutation in some period of (evolutionary) time</h3>
<div class="incremental">
<ul>
<li>Nucleotide models are fit “on the fly”
<ul>
<li>e.g. F81, HKY, GTR</li>
</ul></li>
<li>Protein models are typically pre-made
<ul>
<li>e.g. <a href="http://www.scopus.com/record/display.url?eid=2-s2.0-0026691182&amp;origin=inward&amp;txGid=u2307zIZ5me4AtASSSuc_3u%3a2">JTT</a> (Jones, Taylor, and Thornton), and <a href="http://mbe.oxfordjournals.org/content/18/5/691.short">WAG</a> (Whelan and Goldman) matrices</li>
</ul></li>
<li>Codon models are a great idea
<ul>
<li>Position matters!</li>
<li>e.g. <a href="http://mbe.oxfordjournals.org/content/23/1/7.full">SRD06</a> model</li>
</ul></li>
</ul>
</div>
</section>
<section id="model-hierarchy-from-posada-and-crandall" class="slide level1">
<h3>Model hierarchy, from <a href="http://bioinformatics.oxfordjournals.org/cgi/reprint/14/9/817">Posada and Crandall</a></h3>
<p><img src="figures/phylo/hLRT_Posada_and_Crandal_bioInfo.png" height="620"></p>
</section>
<section id="calculating-likelihood-of-a-single-column" class="slide level1">
<h3>Calculating likelihood of a single column</h3>
<p><img src="figures/phylo/column_likelihood_nofonts.svg" height="620"></p>
</section>
<section id="likelihood-of-an-alignment" class="slide level1">
<h3>Likelihood of an alignment</h3>
<p><img width="770" src="figures/phylo/aln_likelihood.scour_nofonts.svg"></p>
Note assumption of independence between sites!
</section>
<section id="the-phylogenetic-likelihood-of-a-tree-is-the-likelihood-of-generating-the-observed-data-given-that-tree-under-the-sequence-evolution-model" class="slide level1">
<h3>The phylogenetic likelihood of a tree is the likelihood of generating the observed data given that tree (under the sequence evolution model)</h3>
<div class="incremental">
<ul>
<li>Maximum likelihood gives a point estimate</li>
<li>Confidence is assessed using the <a href="http://www.jstor.org/pss/2408678">bootstrap</a></li>
<li>Lots of flexibility with models</li>
</ul>
</div>
</section>
<section id="bayes-is-magic" class="slide level1">
<h3>Bayes is <a href="http://en.wikipedia.org/wiki/Bayes%27_theorem">magic</a></h3>
<p> </p>
<p> </p>
<p> </p>
<p><span class="math display">\[
P(\theta \mid D) \propto P(D \mid \theta) P(\theta)
\]</span></p>
</section>
<section id="the-posterior-probability-of-parameters" class="slide level1">
<h3>The <em>posterior probability</em> of parameters</h3>
<p><img src="figures/like_intro/ripe_posterior_nofonts.svg" height="620"></p>
</section>
<section id="the-posterior-probability-of-a-tree-is-the-probability-that-the-observed-tree-is-correct-given-the-model-and-priors" class="slide level1">
<h3>The posterior probability of a tree is the probability that the observed tree is correct (given the model and priors)</h3>
<ul>
<li>Bayesians sample from this posterior</li>
<li>If you can deal with a prior, it’s the statistically right thing to do</li>
<li>Sometimes we aren’t actually interested in the tree, so we can integrate it out</li>
<li>But! Short alignment, 100 taxa = hours</li>
</ul>
</section>
<section id="markov-chain-monte-carlo" class="slide level1">
<h3>Markov chain Monte Carlo</h3>
<p><img src="figures/sprmix/5-taxa-treespace-mcmc-unrooted_nofonts.svg" height="620"></p>
</section>
<section id="metropolis-hastings-algorithm" class="slide level1">
<h3><a href="http://en.wikipedia.org/wiki/Metropolis–Hastings_algorithm">Metropolis-Hastings algorithm</a></h3>
<ul>
<li>If you jump to a better tree, accept that move</li>
<li>If you jump to a worse tree, accept that move with a non-zero probability</li>
<li>It’s all arranged so that you sample trees in proportion to their posterior probability</li>
</ul>
</section>
<section id="subset-to-high-probability-nodes" class="slide level1">
<h3>Subset to high probability nodes</h3>
<p><img src="figures/sprmix/5-taxa-treespace-subgraph-unrooted_nofonts.svg" height="620"></p>
</section>
<section id="real-tree-spaces-have-bottlenecks" class="slide level1">
<h3>Real tree spaces have bottlenecks</h3>
<img src="figures/sprmix/ds6.png" height="550">
<div class="caption">
Whidden &amp; M, <em>Systematic Biology</em>, 2015
</div>
</section>
<section id="likelihood-phylogenetics-recap" class="slide level1">
<h3>Likelihood phylogenetics recap</h3>
<div class="incremental">
<ul>
<li>In likelihood phylogenetics, explicitly model mutation process</li>
<li>This allows complex models to be used</li>
<li>Statistical basis allows us to make formal statements about uncertainty</li>
<li>But on the other hand our models are over-simple!</li>
</ul>
</div>
</section>
<section id="crazy-but-typical-model-assumptions" class="slide level1">
<h3>Crazy but typical model assumptions</h3>
<ul>
<li>differences between sequences only appear by point mutation</li>
<li>evolution happens on each column independently</li>
<li>sequences are evolving according to <em>reversible</em> models (this excludes selection and directional evolution of base composition)</li>
<li>the evolutionary process is identical on all branches of the tree</li>
</ul>
</section>
<section id="to-read" class="slide level1">
<h3>To read</h3>
<ul>
<li><a href="http://www.sinauer.com/detail.php?id=1775"><em>Inferring Phylogenies</em></a> by Felsenstein</li>
<li><a href="http://www.cambridge.org/gb/knowledge/isbn/item2327447/?site_locale=en_GB"><em>The phylogenetic handbook</em></a> edited by Lemey, Salemi, and Vandamme, chapters by the stars</li>
<li><a href="http://treethinkers.org/tutorials/">Bodega phylogenetics tutorials</a></li>
</ul>
</section>
<section id="software" class="slide level1">
<h3>Software</h3>
<ul>
<li><a href="http://www.microbesonline.org/fasttree/">FastTree</a> – approximate ML</li>
<li><a href="http://wwwkramer.in.tum.de/exelixis/software.html">RAxML</a> and <a href="http://www.atgc-montpellier.fr/phyml/">PhyML</a> – somewhat less approximate ML</li>
<li><a href="http://beast.bio.ed.ac.uk/">BEAST</a> – Bayesian</li>
<li><a href="http://mrbayes.csit.fsu.edu/">MrBayes</a> – Bayesian</li>
<li>many others, but these are the ones I know …</li>
</ul>
</section>
    </div>
  </div>

  <script src="../reveal/lib/js/head.min.js"></script>
  <script src="../reveal/js/reveal.min.js"></script>
  <script src="../reveal/js/config.js"></script>

  </body>
</html>
