<!doctype html>
<html lang="en">


	<head>
		<meta charset="utf-8">

		<title>B-cell selection and evolution</title>

		<meta name="author" content="Sarah Cobey">

		<meta name="apple-mobile-web-app-capable" content="yes" />
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />

		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<link rel="stylesheet" href="../reveal/css/reveal.min.css">
		<link rel="stylesheet" href="../reveal/css/theme/trvrb.css" id="theme">
		<link rel="stylesheet" href="../reveal/css/font-awesome/css/font-awesome.min.css">

		<!-- For syntax highlighting -->
		<link rel="stylesheet" href="../reveal/lib/css/zenburn.css">

		<!--[if lt IE 9]>
		<script src="lib/js/html5shiv.js"></script>
		<![endif]-->
	</head>

	<body>

		<div style="position: absolute; top:10px; left:10px; z-index:100;">
			<a href="http://bedford.io/projects/mitii/bcells/">
				<i class="fa fa-times-circle" style="color: #bbb;"></i>
			</a>
		</div>

		<div class="reveal">

			<!-- Any section element inside of this container is displayed as a slide -->
			<div class="slides">

				<section data-background="#5555CC">
					<h3 class="title">Selection and evolution in B cells</h3>

                    <br>

                    <br>

                    Most slides from <a href="http://cobeylab.uchicago.edu/"> Sarah Cobey </a>; some from Erick
				</section>

				<section>
					<h3>B cells determine antibody diversity</h3>
					<img class="stretch" src="images/bcell.png">
					<div class="citation">
						<a href="http://www.garlandscience.com/product/isbn/9780815342434;jsessionid=jUIKR3YCu+EU9PPmbulBjg__">
							Murphy 2011
						</a>
					</div>
				</section>

				<section>
					<h3>Structure of a B cell receptor</h3>
					<img class="stretch" src="images/bcr.png">
					<div class="citation">
						<a href="http://www.nature.com/nbt/journal/v32/n2/abs/nbt.2782.html">
							Georgiou et al. 2014
						</a>
					</div>
				</section>

				<section>
					<h3>Origins of B cell receptor diversity</h3>
					<p>VDJ recombination
					<p>Affinity maturation

				</section>

				<section>
					<h3>VDJ loci</h3>
					<img class="stretch" src="images/VDJ_loci.png">
					<div class="citation">
						<a href="http://www.garlandscience.com/product/isbn/9780815342434;jsessionid=jUIKR3YCu+EU9PPmbulBjg__">
							Murphy 2011
						</a>
					</div>
				</section>

				<section>
					<h3>VDJ recombination</h3>
					<img class="stretch" src="images/vdj_recombination.png">
						<div class="citation">
							<a href="http://www.garlandscience.com/product/isbn/9780815342434;jsessionid=jUIKR3YCu+EU9PPmbulBjg__">
								Murphy 2011
							</a>
						</div>
						</section>

				<section>
					<h3>Selection on naive repertoire</h3>
					<p>Stability
					<p>Autoreactivity
				</section>

<section data-transition="none">
<video autoplay loop>
<source src="images/victora-movieS3.webm" type="video/webm">
</video>
<div class="citation">
	<a href="http://dx.doi.org/10.1126/science.aad3439">
		Tas et al. 2016
	</a>
</div>
</section>
<section data-transition="none">
<p><img src="images/gc-victora-mesin-1_nofonts.svg" height="620"></p>
</section>
<section data-transition="none">
<p><img src="images/gc-victora-mesin-2_nofonts.svg" height="620"></p>
</section>
<section data-transition="none">
<p><img src="images/gc-victora-mesin-3_nofonts.svg" height="620"></p>
</section>
<section data-transition="none">
<p><img src="images/gc-victora-mesin-4_nofonts.svg" height="620"></p>
</section>
<section data-transition="none">
<p><img src="images/gc-victora-mesin-5_nofonts.svg" height="620"></p>
</section>
<section data-transition="none">
<p><img src="images/gc-victora-mesin-6_nofonts.svg" height="620"></p>
</section>
<section data-transition="none">
<p><img src="images/gc-victora-mesin-7_nofonts.svg" height="620"></p>
</section>
<section data-transition="none">
<p><img src="images/gc-victora-mesin_nofonts.svg" height="620"></p>
</section>
<section data-transition="none">
<p><img src="images/many-gcs_nofonts.svg" height="620"></p>
</section>
<section data-transition="none">
<p><img src="images/gc-trees_nofonts.svg" height="620"></p>
</section>

				<section>
					<h3>Somatic hypermutation is AID + repair</h3>
					<img class="stretch" src="images/aid_pol.png">
					<div class="citation">
						<a href="http://www.annualreviews.org/doi/pdf/10.1146/annurev.genet.41.110306.130340">
							Teng and Papavasiliou 2007
						</a>
					</div>
				</section>

				<section>
					<h3>Mutation rate varies over BCR</h3>
					<img class="stretch" src="images/mutations_bcr.jpg">
						<div class="citation">
							<a href="http://www.pnas.org/content/112/7/E728.full">
								Wei et al. 2015
							</a>
						</div>
				</section>

<section data-transition="none">
<h3>The mutations have peculiar biases</h3>
<p><br></p>
<p><img width="970" src="images/yeap-alt-1_nofonts.svg"></p>
</section>
<section data-transition="none">
<h3>The mutations have peculiar biases</h3>
<p><br></p>
<p><img width="970" src="images/yeap-alt_nofonts.svg"></p>
</section>
<section id="one-would-like-to-quantify-these-biases" class="slide level1">
<h3>One would like to quantify these biases</h3>
<p><img src="images/yaari2015-hedgehog-with-central-g_nofonts.svg" height="620"></p>
</section>

				<section>
                <h3>Phylogenetics is not straightforward</h3>
					<img class="stretch" src="images/hoehn-mean-field.png">
						<div class="citation">
							<a href="http://dx.doi.org/10.1534/genetics.116.196303">
								Hoehn et al. 2017
							</a>
						</div>
				</section>




				<section>
					<h3>GC B cells evolve</h3>
					<img class="stretch" src="images/gc_micro.png">
						<div class="citation">
							<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2119370/pdf/je1763679.pdf">
								Jacob and Kelsoe 1992
							</a>
						</div>
				</section>

				<section>
					<h3>Permissive selection vs complex antigens</h3>
					<img class="stretch" src="images/kuraoka-avidities_nofonts.svg">
						<div class="citation">
							<a href="http://www.sciencedirect.com/science/article/pii/S1074761316300486">
								Kuraoka et al. 2016
							</a>
						</div>
				</section>

				<section>
					<h3>Memory cells can affinity mature</h3>
					<img src="images/andrews_memory.jpg" style="width: 70%; height: 70%;">
						<div class="citation">
							<a href="http://stm.sciencemag.org/content/7/316/316ra192">
								Andrews et al. 2015
							</a>
						</div>
						</section>

				<section>
					<h3>Antibody isotypes</h3>
					<img class="stretch" src="images/isotypes.png">
						<div class="citation">
							<a href="http://absoluteantibody.com/antibody-resources/antibody-overview/antibody-isotypes-subtypes/">
								Absolute Antibody
							</a>
						</div>
				</section>

				<section>
					<h3>Isotype switching</h3>
					<img class="stretch" src="https://upload.wikimedia.org/wikipedia/commons/b/b3/Class_switch_recombination.png">
						<div class="citation">
							<a href="https://en.wikipedia.org/wiki/Immunoglobulin_class_switching">
								Wikipedia
							</a>
						</div>
				</section>

				<section>
					<h3>Clones can be long-lived</h3>
					<img class="stretch" src="images/ch103.png">
						<div class="citation">
							<a href="http://www.nature.com/nature/journal/v496/n7446/full/nature12053.html">
								Liao et al. 2013
							</a>
						</div>
					</section>

				<section>
					<h3>Broadly neutralizing antibodies are often highly diverged</h3>
					<img class="stretch" src="images/ch103_k.png">
						<div class="citation">
							<a href="http://www.nature.com/nature/journal/v496/n7446/full/nature12053.html">
								Liao et al. 2013
							</a>
					</div>
				</section>

				<section>
					<h3>Broadly neutralizing antibodies to flu elusive</h3>
					<img class="stretch" src="images/stalk_bnab.png">
						<div class="citation">
							<a href="http://www.sciencemag.org/content/333/6044/843.figures-only">
								Ekiert et al. 2011
							</a>
						</div>
				</section>

				<section data-background="#99CCFF">
					<p>Why do some people develop broadly neutralizing antibodies?</p>
					<p>Can we induce them in everyone?
					<p>Will they dominate?
				</section>

				<section>
					<h3>"Restricted" responses to epitopes</h3>
					<img class="stretch" src="images/vh_frequencies.png">
						<div class="citation">
							<a href="http://rstb.royalsocietypublishing.org/content/370/1676/20140238.long">
								Dunand and Wilson 2015
							</a>
						</div>
				</section>

				<section>
					<h3>Genotype affects response to H5N1 vaccination</h3>
					<img class="stretch" src="images/avnir_prepost.jpg">
						<div class="citation">
							<a href="http://www.nature.com/articles/srep20842">
								Avnir et al. 2016
							</a>
						</div>
				</section>

				<section>
					<h3>Allele copy number affects clone frequency</h3>
					<img src="images/avnir_igg_frequency.png" style="width: 50%; height: 50%;">
						<div class="citation">
							<a href="http://www.nature.com/articles/srep20842">
								Avnir et al. 2016
							</a>
						</div>
				</section>

				<section>
					<h3>Ethnic differences in copy number</h3>
					<img class="stretch" src="images/cnv_ethnic.png">
						<div class="citation">
							<a href="http://www.nature.com/articles/srep20842">
								Avnir et al. 2016
							</a>
						</div>
                </section>


<section id="section" class="slide level1">
<h1> How can we learn using probabilistic models? </h1>
</section>
<section id="we-dont-need-lots-of-little-techniques" class="slide level1">
We don’t need lots of little techniques…
<p><img src="images/chicobonbon.jpg" height="500"></p>
<p>we only need models and one principle: <em>likelihood</em>.</p>
</section>
<section id="we-can-sequence-bcrs-in-high-throughput" class="slide level1">
<h3>We can sequence BCRs in high throughput</h3>
<p><img src="images/bcell-mess-1_nofonts.svg" height="620"></p>
</section>
<section id="how-do-we-learn-from-bcr-sequences" class="slide level1">
<h3>How do we learn from BCR sequences?</h3>
<p><img src="images/bcell-mess-2_nofonts.svg" height="620"></p>
</section>
<section id="section-1" class="slide level1">
<ol style="font-size:1.4em">
<li>
Develop probabilistic model of biological process and how data is generated.
</li>
<li class="fragment">
Find parameter choices that <em>maximize the likelihood</em> of generating the observed data.
</li>
</ol>
<p><br></p>
<p><br></p>
<div class="fragment">
<p>When a field of bioinformatics is mature, it becomes statistics.<br> <em>E.g.</em> Maximum-likelihood phylogenetics, HMMER, DESeq2, etc.</p>
</div>
</section>
<section id="vdj-rearrangement" class="slide level1">
<h3>VDJ rearrangement</h3>
<p><br></p>
<p><img width="970" src="images/overview-top_nofonts.svg"></p>
</section>
<section id="section-2" class="slide level1">
<p><img src="images/overview-hmm-1_nofonts.svg" height="620"></p>
</section>
<section id="hmm-intro-dishonest-casino" class="slide level1">
<h3>HMM intro: dishonest casino</h3>
<p><img src="images/dishonest-casino-1_nofonts.svg" height="620"></p>
</section>
<section id="hmm-intro-dishonest-casino-1" class="slide level1">
<h3>HMM intro: dishonest casino</h3>
<p><img src="images/dishonest-casino-2_nofonts.svg" height="620"></p>
</section>
<section id="hmm-intro-dishonest-casino-2" class="slide level1">
<h3>HMM intro: dishonest casino</h3>
<p><img src="images/dishonest-casino-3_nofonts.svg" height="620"></p>
</section>
<section id="hmm-intro-dishonest-casino-3" class="slide level1">
<h3>HMM intro: dishonest casino</h3>
<p><img src="images/dishonest-casino_nofonts.svg" height="620"></p>
</section>
<section id="section-3" class="slide level1">
<h3></h3>
<p><img src="images/dishonest-casino-expanded-1_nofonts.svg" height="620"></p>
</section>
<section id="section-4" class="slide level1">
<h3></h3>
<p><img src="images/dishonest-casino-expanded_nofonts.svg" height="620"></p>
</section>
<section id="section-5" class="slide level1">
<h3></h3>
<p><img src="images/dishonest-casino-expanded_nofonts.svg" height="450"></p>
<ol type="1">
<li>What is the likelihood of the data under this path (write equation)</li>
<li>If <span class="math inline">\(p\)</span> is close to 0.5, what is an alternate path that will have a higher likelihood? What is this likelihood?</li>
</ol>
</section>
<section id="section-6" class="slide level1">
<h3></h3>
<p><img src="images/overview-hmm-2_nofonts.svg" height="620"></p>
</section>
<section id="section-7" class="slide level1">
<h3></h3>
<p><img src="images/overview-hmm_nofonts.svg" height="620"></p>
</section>
<section id="section-8" class="slide level1">
<p><img src="images/overview-hmm_nofonts.svg" height="500"></p>
<p>Biased die roll <span class="math inline">\(\leftrightarrow\)</span> read base (under mutation)<br> Switching dice <span class="math inline">\(\leftrightarrow\)</span> changing between V, D, J genes</p>
</section>
<section id="section-9" class="slide level1">
<p><img width="970" src="images/vdj_recomb_inference_nofonts.svg"></p>
<div class="fragment">
<p><img src="images/overview-hmm-1_nofonts.svg" style="float:right" height="300"></p>
<p> </p>
<p><em>VDJ annotation problem:</em> from where did each nucleotide come?</p>
<p>A: Take maximum-likelihood HMM path.</p>
</div>
</section>
<section id="what-are-probabilities" class="slide level1">
<h3>What are probabilities?</h3>
<p><img src="images/overview-hmm_nofonts.svg" height="620"></p>
</section>
<section id="distributions-are-reproducibly-weird" class="slide level1">
<h3>Distributions are reproducibly weird!</h3>
<p><img src="images/IGHV2-70_star_12-v_3p_del_nofonts.svg" height="620"></p>
</section>
<section id="distributions-are-reproducibly-weird-1" class="slide level1">
<h3>Distributions are reproducibly weird!</h3>
<p><a href="images/IGHV2-70_star_12-v_3p_del_nofonts.svg" target="_blank"><img width="380" src="images/IGHV2-70_star_12-v_3p_del_nofonts_nofonts.svg"></a> <a href="images/IGHD1-14_star_01-d_5p_del_nofonts.svg" target="_blank"><img width="380" src="images/IGHD1-14_star_01-d_5p_del_nofonts_nofonts.svg"></a> <a href="images/IGHD7-27_star_01-d_3p_del_nofonts.svg" target="_blank"><img width="380" src="images/IGHD7-27_star_01-d_3p_del_nofonts_nofonts.svg"></a> <a href="images/IGHJ4_star_02-j_5p_del_nofonts.svg" target="_blank"><img width="380" src="images/IGHJ4_star_02-j_5p_del_nofonts_nofonts.svg"></a></p>
<aside class="notes">
<ul>
<li>IGHV2-70 12 . number of reads.</li>
<li>Adaptive: 4794 5987 2181</li>
<li>Stanford: 171 110 553 97 126 210 500 597 151 131 358 81 139 99 105</li>
</ul>
</aside>
</section>
<section id="distributions-are-reproducibly-weird-2" class="slide level1">
<h3>Distributions are reproducibly weird!</h3>
<p><a href="images/IGHV3-23D_star_01_nofonts.svg" target="_blank"><img width="600" src="images/IGHV3-23D_star_01_nofonts_nofonts.svg"></a> <a href="images/IGHV3-33_star_06_nofonts.svg" target="_blank"><img width="600" src="images/IGHV3-33_star_06_nofonts_nofonts.svg"></a></p>
</section>
<section id="find-clonal-families" class="slide level1">
<h3>Find clonal families</h3>
<p><img src="images/clustering_goal_nofonts.svg" height="620"></p>
</section>
<section id="find-clonal-families-1" class="slide level1">
<h3>Find clonal families</h3>
<p><img src="images/clustering_goal_nofonts.svg" height="500"></p>
<p><em>Did two sequences come from the same VDJ recombination?</em></p>
</section>
<section id="say-we-are-given-two-sequences" class="slide level1">
<h3>Say we are given <em>two</em> sequences</h3>
<p><img src="images/dishonest-casino-pair_nofonts.svg" height="620"></p>
</section>
<section id="double-roll-leftrightarrow-pair-hmm" class="slide level1">
<h3>Double roll <span class="math inline">\(\leftrightarrow\)</span> Pair HMM</h3>
<p><img src="images/dishonest-casino-expanded-pair_nofonts.svg" height="550"></p>
</section>
<section id="double-roll-leftrightarrow-pair-hmm-1" class="slide level1">
<h3>Double roll <span class="math inline">\(\leftrightarrow\)</span> Pair HMM</h3>
<p><img src="images/dishonest-casino-expanded-pair_nofonts.svg" style="float:left" height="550"></p>
<p><br></p>
<p><br></p>
<p>Pick double roll hypothesis if it has higher likelihood of generating data.</p>
<p><br></p>
<div class="fragment">
<p>What about BCRs?</p>
</div>
</section>
<section id="section-10" class="slide level1">
<h3></h3>
<p><img width="970" src="images/separate-or-together_nofonts.svg"></p>
<p><br></p>
<div class="fragment">
(But we only know the sequences, not the annotations!)
</div>
</section>
<section id="section-11" class="slide level1">
<p><img src="images/overview-hmm-1_nofonts.svg" height="620"></p>
</section>
<section id="do-two-sequences-come-from-a-single-rearrangement-event" class="slide level1">
<h3>Do two sequences come from a single rearrangement event?</h3>
<p>Probability of generating observed sequence <span class="math inline">\(x\)</span> from HMM:</p>
<p><span class="math display">\[
\mathbb P(x) = \sum_{\text{paths}\ \sigma} \mathbb P(x;\sigma),
\]</span></p>
<div class="fragment">
<p>Probability of generating two sequences <span class="math inline">\(x\)</span> and <span class="math inline">\(y\)</span> from the same path through the HMM (i.e. from the same rearrangement event):</p>
<p><span class="math display">\[
\mathbb P(x,y) = \sum_{\text{paths}\ \sigma} \mathbb P(x,y;\sigma),
\]</span></p>
</div>
<div class="fragment">
<p><span class="math display">\[
\text{Calculate: } \frac{\mathbb P(x, y)}{\mathbb P(x) \mathbb P(y)} = \frac{\mathbb P(\text{single rearrangement})}{\mathbb P(\text{independent rearrangements})}
\]</span></p>
</div>
</section>
<section id="do-sets-of-sequences-come-from-a-single-rearrangement-event" class="slide level1">
<h3>Do sets of sequences come from a <br> single rearrangement event?</h3>
<p> </p>
<p><span class="math display">\[
\frac{\mathbb P(A \cup B)}{\mathbb P(A) \mathbb P(B)} = \frac{\mathbb P(A \cup B \ | \ \text{single rearrangement})}{\mathbb P(A,B  \ | \ \text{independent rearrangements})}
\]</span></p>
<p> </p>
<p>Use this for agglomerative clustering; stop when the ratio &lt; 1.</p>
<p><img width="970" src="images/agglomerative-clustering_nofonts.svg"></p>
</section>

<section>
<ul>
<li> Munshaw, S., & Kepler, T. B. (2010). <em>SoDA2: a Hidden Markov Model approach for identification of immunoglobulin rearrangements. </em> Bioinformatics. </li>
<li>Murugan, Mora, Walczak, & Callan (2012). <em>Statistical inference of the generation probability of T-cell receptors from sequence repertoires.</em> PNAS. </li>
<li>Ralph &amp; M. (2016). <em>Consistency of VDJ Rearrangement and Substitution Parameters Enables Accurate B Cell Receptor Sequence Annotation.</em> PLOS Computational Biology.</li>
<li>Ralph &amp; M. (2016). <em>Likelihood-based inference of B-cell clonal families.</em>PLOS Computational Biology.</li>
<li>
Elhanati, Sethna, Marcou, Callan, Mora, \& Walczak (2015). <em> Inferring processes underlying B-cell repertoire diversity. </em> Philosophical Transactions of the Royal Society of London.
</li>
<li>
Elhanati, Marcou, Mora, & Walczak (2016). <em> repgenHMM: a dynamic programming tool to infer the rules of immune receptor generation from sequence data.<em> Bioinformatics.
</li>
</ul>
</section>


			</div>

		</div>

		<script src="../reveal/lib/js/head.min.js"></script>
		<script src="../reveal/js/reveal.min.js"></script>
		<script src="../reveal/js/config.js"></script>

	</body>
</html>
